ecs-cli/modules/utils/compose/convert_task_definition_test.go

```diff
@@ -114,6 +115,71 @@ func TestConvertToTaskDefinition(t *testing.T) {
 	assert.Equal(t, taskRoleArn, aws.StringValue(taskDefinition.TaskRoleArn), "Expected taskRoleArn to match")
 }
 
+func TestConvertToTaskDefinitionWithECSFields(t *testing.T) {
+
+	ecsFieldsString := `version: 1
+task_definition:
+  ecs_network_mode: host
+  task_role_arn: arn:aws:iam::123456789012:role/my_role`
+
+	content := []byte(ecsFieldsString)
+	tmpfile, err := ioutil.TempFile("", "ecs-fields")
+
+	if err != nil {
```

Comment:
So generally in unit tests I use `assert.NoError`.

For example, `assert.NoError(t, err, "Could not create ecs fields tempfile").

Reply:
Hm, I see that being used elsewhere and understand why it might be useful, but in this case the creation of the tempfile is auxiliary to the main assertion of the test (i.e. it's just set up, and not what the test really cares about). Do we still use `assert.NoError` in those cases? For me I find it to be distracting if we're treating the test as documentation, but open to other ideas.

Reply:
I would still use `assert.NoError` since if any of these fails the test will or should fail. Also, the best practice is to not use `Errorf`, `Wrapf`, etc.

Reply:
Noting here that we discussed the best practices around Errorf/Wrapf yesterday and came to the consensus that we should in fact use `Wrapf`.